rlclientlib/vw_model/vw_model.cc

```diff
@@ -67,6 +67,14 @@ int vw_model::update(const model_data& data, bool& model_ready, api_status* stat
   return error_code::success;
 }
 
+int vw_model::add_lru_dedup_cache(uint64_t hash, std::string action_str, api_status* status)
```

Comment:
We should probably move the "parse action and populate the cache" functionality into `vw_model` and just have a pointer of it in each vw instance

that way whenever you use vw for a rank call we could just do

```
auto vw = _vw_pool.get_or_create()
vw.set_action_cache(&action_cache)
vw.rank(...)
```

Reply:
the way it is done here the specific action is only passed into one of the vw instances in the object pool

Reply:
removed lru_cache from safe_vw and constructor etc. Added as an arg to load_action and rank in safe_vw